codecarbon/core/co2_signal.py

```diff
@@ -0,0 +1,35 @@
+import requests
+
+from codecarbon.core.units import EmissionsPerKwh, Energy
+from codecarbon.external.geography import GeoMetadata
+
+URL = "https://api.co2signal.com/v1/latest"
+CO2_SIGNAL_API_TOKEN = None
+
+
+def is_available():
+    if CO2_SIGNAL_API_TOKEN:
+        return True
+    else:
+        return False
+
+
+def get_emissions(energy: Energy, geo: GeoMetadata):
+    if geo.latitude:
+        params = {"lat": geo.latitude, "lon": geo.longitude}
+    else:
+        params = {"countryCode": geo.country_2letter_iso_code}
+    resp = requests.get(
+        URL, params=params, headers={"auth-token": CO2_SIGNAL_API_TOKEN}
+    )
```

Comment:
Add a timeout and a test for it. You can test it with a very low timeout, like 0.00000001